### PR TITLE
FAI-14379 | Override mode for GitHub / Jira

### PR DESCRIPTION
**Pull request by dbruno21**

### CODE REVIEW

File: sources/jira-source/src/index.ts [hunk 1]
```diff
@@ -44,6 +44,13 @@ export class JiraSource extends AirbyteSourceBase<JiraConfig> {
     return 'jira';
   }
 
+  mode(config: JiraConfig): string | undefined {
+    if (!config.url) {
+      return undefined;
```
Comment (dbruno21) [inline sources/jira-source/src/index.ts:49]: if url not present, throws error on checkConnection

Comment (tovbinm) [inline sources/jira-source/src/index.ts:49]: How can this be no url? Let's assume cloud then.

File: sources/jira-source/src/index.ts [hunk 2]
```diff
@@ -44,6 +44,13 @@ export class JiraSource extends AirbyteSourceBase<JiraConfig> {
...
+    }
+    return config.url.endsWith('.atlassian.net') ? 'cloud' : 'server';
```
Comment (ypc-faros) [inline sources/jira-source/src/index.ts:51]: nit: Better check against [this](https://github.com/faros-ai/airbyte-connectors/blob/13818da80e10a69c28d18c412500d2fbc1850e29/sources/jira-source/src/jira.ts#L118) regex

**Review by ypc-faros — APPROVED** [review]

File: sources/jira-source/src/index.ts (hunk 2 above)
Comment (chalenge) [reply sources/jira-source/src/index.ts:51]: +1

File: sources/jira-source/src/jira.ts [hunk 3]
```diff
@@ -115,7 +115,7 @@ const prRegex = new RegExp(
     'stateCount=(?<count>[0-9]+)})'
 );
 
-const jiraCloudRegex = /^https:\/\/(.*)\.atlassian\.net/g;
+export const JIRA_CLOUD_REGEX = /^https:\/\/(.*)\.atlassian\.net/g;
```
Comment (tovbinm) [inline sources/jira-source/src/jira.ts:118]: what if the url does not end with `/`?

Comment (dbruno21) [reply sources/jira-source/src/jira.ts:118]: There's no `/` at the end, only the regexp delimiter

**Pull request merged**